### ISSUE TITLE
[firebase_ui_auth] Error compiling project

### ISSUE THREAD

**Issue by jahertor** [firebase/FirebaseUI-Flutter#140]
### Is there an existing issue for this?

- [X] I have searched the existing issues and found no duplicates.

### What plugin is this bug for?

Firebase UI Auth

### What platform(s) does this bug affect?

Android, iOS, Web, macOS, Linux, Windows

### List of dependencies used.

<details>
<summary>flutter pub deps -s list</summary>
<pre>
Dart SDK 3.0.2
Flutter SDK 3.10.2
firebase_ui_oauth_example 1.0.0+1

dependencies:
- flutter 0.0.0
  - characters 1.3.0
  - collection 1.17.1
  - js 0.6.7
  - material_color_utilities 0.2.0
  - meta 1.9.1
  - vector_math 2.1.4
  - sky_engine any
- cupertino_icons 1.0.6
- firebase_auth 4.10.1
  - firebase_auth_platform_interface ^6.19.1
  - firebase_auth_web ^5.8.2
  - firebase_core ^2.17.0
  - firebase_core_platform_interface ^4.8.0
  - flutter any
  - meta ^1.8.0
- firebase_core 2.17.0
  - firebase_core_platform_interface ^4.8.0
  - firebase_core_web ^2.8.0
  - flutter any
  - meta ^1.8.0
- firebase_ui_oauth 1.4.12
  - desktop_webview_auth ^0.0.13
  - firebase_auth ^4.10.1
  - firebase_ui_auth ^1.9.0
  - firebase_ui_shared ^1.4.0
  - flutter_svg ^2.0.7
  - flutter any
- firebase_ui_oauth_apple 1.2.12
  - firebase_auth ^4.10.1
  - firebase_ui_oauth ^1.4.12
  - flutter any
- firebase_ui_oauth_facebook 1.2.12
  - firebase_auth ^4.10.1
  - firebase_ui_oauth ^1.4.12
  - flutter any
  - flutter_facebook_auth ^6.0.1
- firebase_ui_oauth_google 1.2.12
  - firebase_auth ^4.10.1
  - firebase_ui_oauth ^1.4.12
  - flutter any
  - google_sign_in ^6.1.0
- firebase_ui_oauth_twitter 1.2.12
  - flutter any
  - firebase_auth ^4.10.1
  - firebase_ui_oauth ^1.4.12
  - twitter_login ^4.3.2

dev dependencies:
- flutter_test 0.0.0
  - flutter any
  - test_api 0.5.1
  - path 1.8.3
  - fake_async 1.3.1
  - clock 1.1.1
  - stack_trace 1.11.0
  - vector_math 2.1.4
  - async 2.11.0
  - boolean_selector 2.1.1
  - characters 1.3.0
  - collection 1.17.1
  - js 0.6.7
  - matcher 0.12.15
  - material_color_utilities 0.2.0
  - meta 1.9.1
  - source_span 1.9.1
  - stream_channel 2.1.1
  - string_scanner 1.2.0
  - term_glyph 1.2.1
- flutter_lints 1.0.4
  - lints ^1.0.0

transitive dependencies:
- _fe_analyzer_shared 61.0.0
  - meta ^1.0.2
- _flutterfire_internals 1.3.7
  - collection ^1.0.0
  - firebase_core ^2.17.0
  - firebase_core_platform_interface ^4.8.0
  - flutter any
  - meta ^1.8.0
- analyzer 5.13.0
  - _fe_analyzer_shared ^61.0.0
  - collection ^1.17.0
  - convert ^3.0.0
  - crypto ^3.0.0
  - glob ^2.0.0
  - meta ^1.7.0
  - package_config ^2.0.0
  - path ^1.8.0
  - pub_semver ^2.0.0
  - source_span ^1.8.0
  - watcher ^1.0.0
  - yaml ^3.0.0
- args 2.4.2
- async 2.11.0
  - collection ^1.15.0
  - meta ^1.1.7
- boolean_selector 2.1.1
  - source_span ^1.8.0
  - string_scanner ^1.1.0
- characters 1.3.0
- clock 1.1.1
- collection 1.17.1
- convert 3.1.1
  - typed_data ^1.3.0
- crypto 3.0.3
  - typed_data ^1.3.0
- desktop_webview_auth 0.0.13
  - crypto ^3.0.3
  - flutter any
  - http ^1.0.0
  - flutter_web_plugins any
  - plugin_platform_interface ^2.1.4
- email_validator 2.1.17
- facebook_auth_desktop 1.0.1
  - flutter any
  - http ^1.0.0
  - flutter_secure_storage ^9.0.0
  - flutter_facebook_auth_platform_interface ^5.0.0
- fake_async 1.3.1
  - clock ^1.1.0
  - collection ^1.15.0
- ffi 2.1.0
- file 7.0.0
  - meta ^1.9.1
  - path ^1.8.3
- firebase_auth_platform_interface 6.19.1
  - _flutterfire_internals ^1.3.7
  - collection ^1.16.0
  - firebase_core ^2.17.0
  - flutter any
  - meta ^1.8.0
  - plugin_platform_interface ^2.1.3
- firebase_auth_web 5.8.2
  - firebase_auth_platform_interface ^6.19.1
  - firebase_core ^2.17.0
  - firebase_core_web ^2.8.0
  - flutter any
  - flutter_web_plugins any
  - http_parser ^4.0.0
  - js ^0.6.3
  - meta ^1.8.0
- firebase_core_platform_interface 4.8.0
  - collection ^1.0.0
  - flutter any
  - flutter_test any
  - meta ^1.8.0
  - plugin_platform_interface ^2.1.3
- firebase_core_web 2.8.0
  - firebase_core_platform_interface ^4.8.0
  - flutter any
  - flutter_web_plugins any
  - js ^0.6.3
  - meta ^1.8.0
- firebase_dynamic_links 5.3.7
  - firebase_core ^2.17.0
  - firebase_core_platform_interface ^4.8.0
  - firebase_dynamic_links_platform_interface ^0.2.6+7
  - flutter any
  - meta ^1.8.0
  - plugin_platform_interface ^2.1.3
- firebase_dynamic_links_platform_interface 0.2.6+7
  - _flutterfire_internals ^1.3.7
  - firebase_core ^2.17.0
  - flutter any
  - meta ^1.8.0
  - plugin_platform_interface ^2.1.3
- firebase_ui_auth 1.9.0
  - email_validator ^2.1.17
  - firebase_auth ^4.10.1
  - firebase_core ^2.17.0
  - firebase_dynamic_links ^5.3.4
  - firebase_ui_localizations ^1.7.0
  - firebase_ui_oauth ^1.4.12
  - firebase_ui_shared ^1.4.0
  - flutter any
  - flutter_localizations any
- firebase_ui_localizations 1.7.0
  - flutter any
  - flutter_localizations any
  - path ^1.8.2
- firebase_ui_shared 1.4.0
  - flutter any
- flutter_facebook_auth 6.0.2
  - flutter any
  - flutter_facebook_auth_platform_interface ^5.0.0
  - flutter_facebook_auth_web ^5.0.0
  - facebook_auth_desktop ^1.0.1
- flutter_facebook_auth_platform_interface 5.0.0
  - flutter any
  - plugin_platform_interface ^2.1.4
- flutter_facebook_auth_web 5.0.0
  - flutter any
  - flutter_web_plugins any
  - js ^0.6.3
  - flutter_facebook_auth_platform_interface ^5.0.0
- flutter_localizations 0.0.0
  - flutter any
  - intl 0.18.0
  - characters 1.3.0
  - clock 1.1.1
  - collection 1.17.1
  - js 0.6.7
  - material_color_utilities 0.2.0
  - meta 1.9.1
  - path 1.8.3
  - vector_math 2.1.4
- flutter_secure_storage 9.0.0
  - flutter any
  - flutter_secure_storage_linux ^1.1.3
  - flutter_secure_storage_macos ^3.0.0
  - flutter_secure_storage_platform_interface ^1.0.1
  - flutter_secure_storage_web ^1.1.1
  - flutter_secure_storage_windows ^3.0.0
  - meta ^1.3.0
- flutter_secure_storage_linux 1.2.0
  - flutter any
  - flutter_secure_storage_platform_interface ^1.0.1
- flutter_secure_storage_macos 3.0.1
  - flutter any
  - flutter_secure_storage_platform_interface ^1.0.1
- flutter_secure_storage_platform_interface 1.0.2
  - flutter any
  - plugin_platform_interface ^2.0.1
- flutter_secure_storage_web 1.1.2
  - flutter any
  - flutter_secure_storage_platform_interface ^1.0.1
  - flutter_web_plugins any
  - js ^0.6.3
- flutter_secure_storage_windows 3.0.0
  - ffi ^2.0.0
  - flutter any
  - flutter_secure_storage_platform_interface ^1.0.1
  - path ^1.8.0
  - path_provider ^2.0.0
  - win32 ^5.0.0
- flutter_svg 2.0.7
  - flutter any
  - vector_graphics ^1.1.7
  - vector_graphics_codec ^1.1.7
  - vector_graphics_compiler ^1.1.7
- flutter_web_plugins 0.0.0
  - flutter any
  - js 0.6.7
  - characters 1.3.0
  - collection 1.17.1
  - material_color_utilities 0.2.0
  - meta 1.9.1
  - vector_math 2.1.4
- glob 2.1.2
  - async ^2.5.0
  - collection ^1.15.0
  - file >=6.1.3 <8.0.0
  - path ^1.8.0
  - string_scanner ^1.1.0
- google_identity_services_web 0.2.1+1
  - js ^0.6.4
  - meta ^1.3.0
- google_sign_in 6.1.5
  - flutter any
  - google_sign_in_android ^6.1.0
  - google_sign_in_ios ^5.5.0
  - google_sign_in_platform_interface ^2.4.0
  - google_sign_in_web ^0.12.0
- google_sign_in_android 6.1.20
  - flutter any
  - google_sign_in_platform_interface ^2.2.0
- google_sign_in_ios 5.6.4
  - flutter any
  - google_sign_in_platform_interface ^2.2.0
  - pigeon ^11.0.1
- google_sign_in_platform_interface 2.4.2
  - flutter any
  - plugin_platform_interface ^2.1.0
  - quiver ^3.0.0
- google_sign_in_web 0.12.0+4
  - flutter any
  - flutter_web_plugins any
  - google_identity_services_web ^0.2.1
  - google_sign_in_platform_interface ^2.4.0
  - http >=0.13.0 <2.0.0
  - js ^0.6.3
- http 1.1.0
  - async ^2.5.0
  - http_parser ^4.0.0
  - meta ^1.3.0
- http_parser 4.0.2
  - collection ^1.15.0
  - source_span ^1.8.0
  - string_scanner ^1.1.0
  - typed_data ^1.3.0
- intl 0.18.0
  - clock ^1.1.0
  - meta ^1.0.2
  - path ^1.8.0
- js 0.6.7
  - meta ^1.7.0
- lints 1.0.1
- matcher 0.12.15
  - async ^2.10.0
  - meta ^1.8.0
  - stack_trace ^1.10.0
  - term_glyph ^1.2.0
  - test_api ^0.5.0
- material_color_utilities 0.2.0
- meta 1.9.1
- package_config 2.1.0
  - path ^1.8.0
- path 1.8.3
- path_parsing 1.0.1
  - vector_math ^2.1.0
  - meta ^1.3.0
- path_provider 2.1.1
  - flutter any
  - path_provider_android ^2.1.0
  - path_provider_foundation ^2.3.0
  - path_provider_linux ^2.2.0
  - path_provider_platform_interface ^2.1.0
  - path_provider_windows ^2.2.0
- path_provider_android 2.2.0
  - flutter any
  - path_provider_platform_interface ^2.1.0
- path_provider_foundation 2.3.1
  - flutter any
  - path_provider_platform_interface ^2.1.0
- path_provider_linux 2.2.1
  - ffi >=1.1.2 <3.0.0
  - flutter any
  - path ^1.8.0
  - path_provider_platform_interface ^2.1.0
  - xdg_directories >=0.2.0 <2.0.0
- path_provider_platform_interface 2.1.1
  - flutter any
  - platform ^3.0.0
  - plugin_platform_interface ^2.1.0
- path_provider_windows 2.2.1
  - ffi ^2.0.0
  - flutter any
  - path ^1.8.0
  - path_provider_platform_interface ^2.1.0
  - win32 >=2.1.0 <6.0.0
- petitparser 5.4.0
  - meta ^1.9.0
- pigeon 11.0.1
  - analyzer ^5.13.0
  - args ^2.1.0
  - collection ^1.15.0
  - meta ^1.7.0
  - path ^1.8.0
  - yaml ^3.1.1
- platform 3.1.3
- plugin_platform_interface 2.1.6
  - meta ^1.3.0
- pub_semver 2.1.4
  - collection ^1.15.0
  - meta ^1.3.0
- quiver 3.2.1
  - matcher ^0.12.10
- sky_engine 0.0.99
- source_span 1.9.1
  - collection ^1.15.0
  - path ^1.8.0
  - term_glyph ^1.2.0
- stack_trace 1.11.0
  - path ^1.8.0
- stream_channel 2.1.1
  - async ^2.5.0
- string_scanner 1.2.0
  - source_span ^1.8.0
- term_glyph 1.2.1
- test_api 0.5.1
  - async ^2.5.0
  - boolean_selector ^2.1.0
  - collection ^1.15.0
  - meta ^1.3.0
  - source_span ^1.8.0
  - stack_trace ^1.10.0
  - stream_channel ^2.1.0
  - string_scanner ^1.1.0
  - term_glyph ^1.2.0
  - matcher >=0.12.15 <0.12.16
- twitter_login 4.4.2
  - crypto ^3.0.1
  - flutter any
  - http ^1.0.0
- typed_data 1.3.2
  - collection ^1.15.0
- vector_graphics 1.1.9
  - flutter any
  - vector_graphics_codec 1.1.9
- vector_graphics_codec 1.1.9
- vector_graphics_compiler 1.1.9
  - args ^2.3.0
  - meta ^1.7.0
  - path_parsing ^1.0.1
  - xml ^6.3.0
  - vector_graphics_codec 1.1.9
  - path ^1.8.0
- vector_math 2.1.4
- watcher 1.1.0
  - async ^2.5.0
  - path ^1.8.0
- win32 5.0.9
  - ffi ^2.1.0
- xdg_directories 1.0.3
  - meta ^1.3.0
  - path ^1.8.0
- xml 6.3.0
  - collection ^1.17.0
  - meta ^1.9.0
  - petitparser ^5.4.0
- yaml 3.1.2
  - collection ^1.15.0
  - source_span ^1.8.0
  - string_scanner ^1.1.0
</pre>
</details>

### Steps to reproduce

1. Download source code
2. Open the example project `packages\firebase_ui_oauth\example`
3. Compile

### Expected Behavior

App should work normally

### Actual Behavior

The app doesn't compile at all. I was trying to integrate Firebase Auth UI in my own project, but I had the same error. Then, I tried downloading source code of the project and the result is the same:

```
/[USER_FOLDER]/AppData/Local/Pub/Cache/hosted/pub.dev/firebase_ui_auth-1.9.0/lib/src/screens/profile_screen.dart:223:25: Error: The method 'showAdaptiveDialog' isn't defined for the class '_LinkedProvidersRowState'.
 - '_LinkedProvidersRowState' is from 'package:firebase_ui_auth/src/screens/profile_screen.dart' ('/[USER_FOLDER]/AppData/Local/Pub/Cache/hosted/pub.dev/firebase_ui_auth-1.9.0/lib/src/screens/profile_screen.dart').
Try correcting the name to the name of an existing method, or defining a method named 'showAdaptiveDialog'.
      confirmed = await showAdaptiveDialog<bool?>(
                        ^^^^^^^^^^^^^^^^^^
/[USER_FOLDER]/AppData/Local/Pub/Cache/hosted/pub.dev/firebase_ui_shared-1.4.0/lib/src/universal_alert.dart:53:24: Error: Member not found: 'AlertDialog.adaptive'.
    return AlertDialog.adaptive(
                       ^^^^^^^^
Failed to compile application.
```

**Comment by mmcquin** [comment]
This occurred to me today as well.

**Comment by mmcquin** [comment]
@jahertor I was able to get past this by explicitly downgrading to `firebase_ui_shared: 1.3.1` in pubspec, which also requires downgrading to `firebase_ui_auth: ^1.8.0`. Still obviously an issues, but in case anyone needs the workaround - this should suffice.

**Comment by darshankawar** [comment]
Thanks for the update. Seeing the same compile errors as reported. Above workaround works.

**Comment by lesnitsky** [comment]
`AlertDialog.adaptive` is available in Flutter 3.13, but I will be making a change to make `UniversalAlert` backward compatible with Flutter 3.10 as well.
